src/_RegistryProvider.ts

```diff
@@ -0,0 +1,77 @@
+import Promise from 'dojo-core/Promise';
+
+import {
+	ActionLike,
+	CombinedRegistry,
+	StoreLike,
+	WidgetLike
+} from './createApp';
+
+/**
+ * Registry to (asynchronously) get instances by their ID.
+ */
+export interface Registry<T> {
```

Comment:
Not that this is critical, but TypeScript allows the implementation of interfaces in classes... if you are using a `class` structure above, _IdentityRegistry, does it make sense to locate this someplace (a `.d.ts` file) that can then be imported for the IdentityRegistry to implement?